addons/mail/models/mail_notification.py

```diff
@@ -1,77 +1,79 @@
-            ], string='Failure type')
+        ("SMTP", "Connection failed (outgoing mail server problem)"),
+        ("RECIPIENT", "Invalid email address"),
+        ("BOUNCE", "Email address rejected by destination"),
+        ("UNKNOWN", "Unknown error"),
+        ], string='Failure type')
     failure_reason = fields.Text('Failure reason', copy=False)
-    read_date = fields.Datetime('Read Date', copy=False)
 
     _sql_constraints = [
         # email notification;: partner is required
         ('notification_partner_required',
-            "CHECK(notification_type NOT IN ('email', 'inbox') OR res_partner_id IS NOT NULL)",
-            'Customer is required for inbox / email notification'),
+         "CHECK(notification_type NOT IN ('email', 'inbox') OR res_partner_id IS NOT NULL)",
+         'Customer is required for inbox / email notification'),
     ]
 
     def init(self):
-        self._cr.execute('SELECT indexname FROM pg_indexes WHERE indexname = %s', ('mail_notification_res_partner_id_is_read_notification_status_mail_message_id',))
+        self._cr.execute('SELECT indexname FROM pg_indexes WHERE indexname = %s',
+                         ('mail_notification_res_partner_id_is_read_notification_status_mail_message_id',))
         if not self._cr.fetchone():
-            self._cr.execute('CREATE INDEX mail_notification_res_partner_id_is_read_notification_status_mail_message_id ON mail_message_res_partner_needaction_rel (res_partner_id, is_read, notification_status, mail_message_id)')
-
-    def format_failure_reason(self):
-        self.ensure_one()
-        if self.failure_type != 'UNKNOWN':
-            return dict(type(self).failure_type.selection).get(self.failure_type, _('No Error'))
-        else:
-            return _("Unknown error") + ": %s" % (self.failure_reason or '')
+            self._cr.execute("""
+                CREATE INDEX mail_notification_res_partner_id_is_read_notification_status_mail_message_id
+                          ON mail_message_res_partner_needaction_rel (res_partner_id, is_read, notification_status, mail_message_id)
+            """)
 
     @api.model_create_multi
     def create(self, vals_list):
         for vals in vals_list:
             if vals.get('is_read'):
                 vals['read_date'] = fields.Datetime.now()
-        return super(Notification, self).create(vals_list)
+        return super(MailNotification, self).create(vals_list)
 
     def write(self, vals):
         if vals.get('is_read'):
             vals['read_date'] = fields.Datetime.now()
-        return super(Notification, self).write(vals)
+        return super(MailNotification, self).write(vals)
+
+    def format_failure_reason(self):
+        self.ensure_one()
+        if self.failure_type != 'UNKNOWN':
+            return dict(type(self).failure_type.selection).get(self.failure_type, _('No Error'))
+        else:
+            return _("Unknown error") + ": %s" % (self.failure_reason or '')
```

Comment:
A side note that can be changed after (and it's probably not the only place in our code where we do it) but `: %s` in strings must be translated too as part of the rest of the text because in some languages they are done differently: already between French and English we add a space before the `:` char and they don't, and I think this phrase structure doesn't even make sense in some other languages.

Reply:
Indeed, there are still a lot of string using that formatting. However here I am just moving code, maybe will do a commit in another branch.